tests/e2e/test_requests.py

```diff
@@ -376,3 +376,27 @@ def request_attributes(method, horizon):
 
     data = np.ndarray(metadata['shape'], metadata['format'], data)
     return metadata, data
+
+
+def gen_sas():
+    return generate_container_signature(STORAGE_ACCOUNT_NAME, CONTAINER, STORAGE_ACCOUNT_KEY)
+
+@pytest.mark.parametrize("path, payload", [
+    ("slice"    ,   make_slice_request()      ),
+    ("fence"    ,   make_fence_request()      ),
+    ("metadata" ,   make_metadata_request()   ),
+    ("horizon"  ,   make_attribute_request()  ),
+])
+@pytest.mark.parametrize("vds, sas, expected", [
+    (SAMPLES10_URL                  , gen_sas() , http.HTTPStatus.OK            ),
+    (f'{SAMPLES10_URL}?{gen_sas()}' , ''        , http.HTTPStatus.OK            ),
+    (f'{SAMPLES10_URL}?invalid_sas' , gen_sas() , http.HTTPStatus.OK            ),
+    (SAMPLES10_URL                  , ''        , http.HTTPStatus.BAD_REQUEST   )
+])
+def test_sas_token_in_url(path, payload, vds, sas, expected):
```

Comment:
Do those tests bring something new, not already tested in `request_test.go`?

Other tests here test something that only Azure can verify. Our policy is to test in E2E only limited number of things that can't be verified anywhere else and have major impact.

I would argue that
- case 1. is a positive one and is already tested in other tests
- cases 2 and 3 are minor details thus better suited to request_test.go
- case 4 may be now added to `test_assure_no_unauthorized_access` as we now allow for empty sas.

Reply:
Yeah! They both are testing almost the same thing but in ´request_test.go´ we are only doing functional tests. But here we are testing the behavior of the endpoint with the real SAS value. As it is changing the endpoint request parameters and providing an alternate approach calling the endpoint, it's good to have a test like this.

- we can skip case 1 but 2 and 3 are needed because we are verifying whether the correct SAS token is used or not.